misoc/targets/kasli.py

```diff
@@ -34,25 +36,61 @@ def __init__(self, cd, async_reset):
         ]
 
 
-class _CRG(Module):
-    def __init__(self, platform):
+class _CRG(Module, AutoCSR):
+    def __init__(self, platform, freq=125.0e6):
         self.clock_domains.cd_sys = ClockDomain()
         self.clock_domains.cd_sys4x = ClockDomain(reset_less=True)
         self.clock_domains.cd_sys4x_dqs = ClockDomain(reset_less=True)
         self.clock_domains.cd_clk200 = ClockDomain()
 
-        clk125 = platform.request("clk125_gtp")
-        platform.add_period_constraint(clk125, 8.)
-        self.clk125_buf = Signal()
-        self.clk125_div2 = Signal()
+        self.clock_sel = CSRStorage()
+        self.pll_locked = CSRStatus()
+        self.mmcm_locked = CSRStatus()
+
+        # old "sys" clock, now bootstrap
+        bootstrap = platform.request("clk125_gtp")
+        platform.add_period_constraint(bootstrap, 8.)
+        bootstrap_buf = Signal()
+        bootstrap_div2 = Signal()
         self.specials += Instance("IBUFDS_GTE2",
             i_CEB=0,
-            i_I=clk125.p, i_IB=clk125.n,
-            o_O=self.clk125_buf,
-            o_ODIV2=self.clk125_div2)
+            i_I=bootstrap.p, i_IB=bootstrap.n,
```

Comment:
You don't need to rename this, it could still be called clk125 to match the hw and migen pin names.